src/test/java/com/systelab/seed/patient/PatientResourceTest.java

```diff
@@ -53,24 +52,67 @@ private Patient getPatientData(String name, String surname, String email, String
+        Patient patient = getPatientData(name, surname, email, medicalNumber);
+        patient.setDob(dob);
+        patient.getAddress().setStreet(street);
+        patient.getAddress().setCity(city);
+        patient.getAddress().setZip(zip);
+        return patient;
+    }
+
     @Description("Create a new patient with name, surname, email and medical number")
     @Test
     public void testCreatePatient() {
-        Patient patient = getPatientData("John", "Burrows", "jburrows@werfen.com", "112233");
+        String expectedName = "John";
+        String expectedSurname = "Burrows";
+        String expectedEmail = "jburrows@werfen.com";
+        String expectedMedNumber = "112233";
+        LocalDate expectedDob = LocalDate.of(1948, 8, 11);
+        String expectedStreet = "E-Street, 90";
+        String expectedCity = "Barcelona";
+        String expectedZip = "08021";
+        Patient patient = getPatientData(expectedName, expectedSurname, expectedEmail, expectedMedNumber);
         Patient patientCreated = given().body(patient)
                 .when().post("/patients/patient")
                 .then().assertThat().statusCode(200)
                 .extract().as(Patient.class);
         TestUtil.checkObjectIsNotNull("Patient", patientCreated);
-        TestUtil.checkField("Name", "John", patientCreated.getName());
-        TestUtil.checkField("Surname", "Burrows", patientCreated.getSurname());
-        TestUtil.checkField("Email", "jburrows@werfen.com", patientCreated.getEmail());
-        // include validation all patient fields
-        TestUtil.checkField("Medical Number", "112233", patientCreated.getMedicalNumber());
-        TestUtil.checkField("DOB", LocalDate.of(1948, 8, 11), patientCreated.getDob());
-        TestUtil.checkField("Street", "E-Street, 90", patientCreated.getAddress().getStreet());
-        TestUtil.checkField("City", "Barcelona", patientCreated.getAddress().getCity());
-        TestUtil.checkField("Zip", "08021", patientCreated.getAddress().getZip());
+        TestUtil.checkField("Name", expectedName, patientCreated.getName());
+        TestUtil.checkField("Surname", expectedSurname, patientCreated.getSurname());
+        TestUtil.checkField("Email", expectedEmail, patientCreated.getEmail());
+        TestUtil.checkField("Medical Number", expectedMedNumber, patientCreated.getMedicalNumber());
+        TestUtil.checkField("DOB", expectedDob, patientCreated.getDob());
+        TestUtil.checkField("Street", expectedStreet, patientCreated.getAddress().getStreet());
+        TestUtil.checkField("City", expectedCity, patientCreated.getAddress().getCity());
+        TestUtil.checkField("Zip", expectedZip, patientCreated.getAddress().getZip());
+    }
+
+    @Description("Create a new patient with name, surname, email, medical number, dob, and complete address")
+    @Test
+    public void testCreatePatientWithAllInfo() {
```

Comment:
We could also create valid and invalid Patients, using boundary field values, as example of boundary values examples.
(This is supposing that current fields implement some valid ranges that we could test..)

Reply:
Provided that the REST API is well documented and reflects the complete specification of the protocol we want to verify. Totally agree with that.

Reply:
I agree on this.

Reply:
> We could also create valid and invalid Patients, using boundary field values, as example of boundary values examples.
> (This is supposing that current fields implement some valid ranges that we could test..)

There is an specific method for that (testCreateInvalidPatient())